jujugui/static/gui/src/app/components/deployment-flow/deployment-flow.js

```diff
@@ -299,7 +329,19 @@ YUI.add('deployment-flow', function() {
       if (this.state.sshKey) {
         args.config = {'authorized-keys': this.state.sshKey};
       }
-      this.props.deploy(this._handleClose, true, this.state.modelName, args);
+      const deploy = this.props.deploy.bind(
+        this, this._handleClose, true, this.state.modelName, args);
+      if (this.state.newTerms.length > 0) {
+        this.props.addAgreement(this.state.termsList, (error, response) => {
+          if (error) {
+            console.error('Could not agree to terms:', error);
```

Comment:
Seems like we need to percolate addNotification into the component so we can display an error to the user here.